core/process.py

```diff
@@ -494,8 +495,16 @@ def Exec(self, arg_vec, environ):
     try:
       os_.execvpe(argv[0], argv, environ)
     except OSError as e:
-      # TODO: Run with /bin/sh when ENOEXEC error (noshebang).  Because all
-      # shells do it.
+      # Run with /bin/sh when ENOEXEC error (no shebang). Because all shells do it.
+      if e.errno == errno.ENOEXEC:
+        _, realpath = _ResolveFile(argv[0], environ["PATH"].split(':'))
+        # Check if file was deleted in the meantime
+        if realpath is None:
+          e.errno = errno.ENOENT
```

Comment:
I think it's clearer to do sys.exit(127) here than to mutate the exception object, if that's the intention.

Reply:
If we exit immediately, we don't print the error message that the file was missing. Especially if we add suggestions for misspelled files, it would be nice not to duplicate that logic.